### PR TITLE
Internationalization

### PR DESCRIPTION
**Pull request by NateScarlet**
qq 20180308111508

* Add translator for other language.
* Add chinese translation.
* Improve unicode support.

### CODE REVIEW

**Comment by mottosso** [comment]
Thanks @NateScarlet, this is really great work. I haven't worked with internationalisation before but implementation-wise this is easy to follow and is very well written.

Give me a day or so to try this out on my end. Is there anything else you are planning on adding or changing?

File: pyblish_lite/i18n/zh_CN.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        <source>Comment..</source>
+        <translation>备注..</translation>
+    </message>
+    <message>
+        <location filename="../window.py" line="722"/>
+        <source>Processing</source>
+        <translation>处理</translation>
+    </message>
+    <message>
+        <location filename="../window.py" line="877"/>
```
Comment (mottosso) [inline pyblish_lite/i18n/zh_CN.ts]: As someone who hasn't written these before, how does one make these? How do we update them? It looks like a line number in there, does that mean that if we refactor the code, we should update this file? Or is this file generated from something else? How does one provide translations for yet another language?

What is the `pyblish_lite/i18n/zh_CN.qm` file? Is there any documentation you could recommend to better understand how this works? Thanks again, so exciting!

Comment (NateScarlet) [reply pyblish_lite/i18n/zh_CN.ts]: The `.ts` file is generated by  `pyside-lupdate pysite_lite.pro`.
The line numbers are just reference for translation, will be removed in .qm file.
Qt matches translation with context(class name), not line number.

The `.qm` file is generated by `lrealese <.ts file>` .

Add another language by doing follow, tools can be found in PySide pacakage folder.:
1. Add `<Locale name>.ts` to 'TRANSLATIONS' of `pyblish_lite.pro`, seperate by space.
  Locale name is return value of `PySide.QtCore.QLocale.system().name()` on you target machine.
2. Run `pyside-lupdate pyblish_lite.pro`
3. Open `.ts` file with `linguist`
4. Translate all items in `linguist`.
5. Release by `lingust` -> `File` menu -> `release`, or use `lrealese <.ts file>`



see https://wiki.qt.io/PySide_Internationalization
http://doc.qt.io/qt-5/i18n-source-translation.html
http://doc.qt.io/qt-5/linguist-translators.html

**Comment by NateScarlet** [comment]
Maybe a filter on terminal, because i am logging all server communication on debug level..

**Comment by mottosso** [comment]
> Maybe a filter on terminal, because i am logging all server communication on debug level..

Yes, sounds like a fine idea. Pyblish QML implements one, ideally it'd look and behave somewhat similarly. This is a discussion well suited for a separate issue or PR.

**Comment by mottosso** [comment]
Ah, now I see what you mean by filter. Looking good.

I've tested it on my end, and it runs fine. Let me know when you're ready, and if you can also increment the version for this, to `0.8.0`, I'll be able to make a release following a merge.

Thanks again @NateScarlet!

**Comment by NateScarlet** [comment]
Filter widget is a lot of work for me, so i just choose the simple way. :)

**Comment by mottosso** [comment]
Yes, it's a clever and minimalist way of achieving it. I like it. :)

**Comment by mottosso** [comment]
@NateScarlet thanks again for this PR, is this a good time to merge?

**Comment by NateScarlet** [comment]
@mottosso Yes, i think there is no more things to fix for now.

**Comment by mottosso** [comment]
Thanks again, putting together a release now.

Could you let me know how to use it? How do I choose Chinese for my language?

**Comment by NateScarlet** [comment]
@mottosso this version let Qt to detect system locale. You can change your system language to chinese, or change code on https://github.com/pyblish/pyblish-lite/blob/65823b469a6e73496aa01a534c0806f3d6ecfaaf/pyblish_lite/app.py#L32 from `QtCore.QLocale.system()` to `QtCore.QLocale('zh_CN')`

**Comment by mottosso** [comment]
Excellent, thanks. Release is [here](https://github.com/pyblish/pyblish-lite/releases/tag/0.8.0), should also appear on PyPI shortly.

**Pull request merged**